### PR TITLE
Fix handling of `throw_on_missing` flag for `select()` on ListConfig

### LINKED ISSUE #563: `select()` ignores `throw_on_missing` for ListConfig
**Describe the bug**
The following code should raise an exception, but does not:

```python
cfg = OmegaConf.create(["???"])
OmegaConf.select(cfg, key, default=0, throw_on_missing=True)
```

### PR DESCRIPTION
**Pull request by odelalleau**
Fixes #563

### CODE REVIEW

File: news/563.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Selecting a missing node from a ListConfig with `throw_on_missing` set to True now raises the intended exception.
```
Comment (omry) [inline news/563.bugfix:1]: Function name here to be more specific about what you mean by "selecting"
```suggestion
OmegaConf.select() a missing node from a ListConfig with `throw_on_missing` set to True now raises the intended exception.
```

File: tests/test_select.py [hunk 2]
```diff
@@ -85,7 +86,7 @@ def test_select_default(
         pytest.param({"missing": "???"}, "missing", id="missing"),
     ],
 )
-def test_select_default_throw_on_missing(
+def test_select_default_throw_on_missing_dict(
```
Comment (omry) [inline tests/test_select.py:89]: old function name seems better. why rename it?

File: tests/test_select.py [hunk 3]
```diff
@@ -99,6 +100,23 @@ def test_select_default_throw_on_missing(
...
+    cfg: Any,
+    key: Any,
+) -> None:
+    cfg = OmegaConf.create(cfg)
+
+    # throw on missing still throws if default is provided
+    with pytest.raises(MissingMandatoryValue):
+        OmegaConf.select(cfg, key, default=0, throw_on_missing=True)
+
+
```
Comment (omry) [inline tests/test_select.py:119]: looks like this is covered by the first test you changed?

**Review by omry — CHANGES_REQUESTED** [review]

**Review by Jasha10 — APPROVED** [review]
Looks good.

File: tests/test_select.py (hunk 2 above)
Comment (Jasha10) [reply tests/test_select.py:89]: Before there was just the `test_select_default_throw_on_missing` function;
now there is a `test_select_default_throw_on_missing_dict` function and a `test_select_default_throw_on_missing_list` function.

File: tests/test_select.py (hunk 3 above)
Comment (Jasha10) [reply tests/test_select.py:119]: I think that this test covers the `throw_on_missing=True` case, the other one does not.

Comment (odelalleau) [reply tests/test_select.py:119]: No, the `list:missing` test I added is a success case when `throw_on_missing` is False (it was working before as well, but it wasn't tested, I just added it for better test coverage)

File: news/563.bugfix (hunk 1 above)
Comment (odelalleau) [reply news/563.bugfix]: Done in b9a5db6

File: tests/test_select.py (hunk 2 above)
Comment (omry) [reply tests/test_select.py:89]: this function is testing both dicts and lists.

File: tests/test_select.py (hunk 3 above)
Comment (omry) [reply tests/test_select.py:119]: How is it a success case if the test body has this?
```
with pytest.raises(MissingMandatoryValue):
   ...
```

**Review by omry — CHANGES_REQUESTED** [review]

File: tests/test_select.py (hunk 2 above)
Comment (odelalleau) [reply tests/test_select.py]: It was only testing a dict here, but actually you're right that I can re-use the same test for lists as well (somehow I thought that `set_struct` was only for DictConfig, but looks like I was wrong).

I changed it in cef0a23

File: tests/test_select.py (hunk 3 above)
Comment (odelalleau) [reply tests/test_select.py]: Trying to clear up the confusion:
* I thought by "first test you changed" you meant `test_select()`: I indeed added a new test here (id `list:missing`) to test the case where we select a missing value with `throw_on_missing` set to False (which worked before, and still works here)
* The test with `with pytest.raises(MissingMandatoryValue):` is `test_select_default_throw_on_missing`: this tests the case where `throw_on_missing` is True. The new test I added here would have failed before this PR.

**Review by omry — APPROVED** [review]
Looking good.
I think GitHub diff view played some tricks on all of us here.

**Pull request merged**